### ISSUE TITLE
Did you use test set for parameter tuning?

### ISSUE THREAD

**Issue by Leavingseason** [geek-ai/irgan#5]
After reading your code (item recommendation), I have several questions, and the first one maybe how you tune the best parameters. In the code you seems to pick the best one according to the test set. However, this will easily leads to overfitting.

**Comment by wnzhang** [comment]
Training/validation sets help researchers to acquire good hyperparameters. From fig 2,3 and 6 we can observe the stable outperformance of IRGAN over the baseline given our chosen hyperparameters, which are not tuned by the test set.

**Comment by andrewyates** [comment]
Did you use the test set to select the best epoch?

https://github.com/geek-ai/irgan/blob/master/ltr-gan/ltr-gan-pointwise/ltr_gan_d_nn_g_nn.py#L152
